mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -72,6 +72,33 @@ where
             bucket_name: bucket_name.to_owned(),
         }
     }
+
+    pub async fn verify_cache_valid(&self) -> ObjectClientResult<(), PutObjectError, Client::ClientError> {
```

Comment:
nit: we can now remove the [todo](https://github.com/awslabs/mountpoint-s3/blob/7a265f5cfd20e6ce2ba76b18785d3b7a75e339c8/mountpoint-s3/src/data_cache/express_data_cache.rs#L66)